src/d3a/setup/area_events/disconnect_interval_event.py

```diff
@@ -0,0 +1,50 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from d3a.models.appliance.simple import SimpleAppliance
+from d3a.models.appliance.switchable import SwitchableAppliance
+from d3a.models.area import Area
+from d3a.models.area.events import DisconnectIntervalAreaEvent
+from d3a.models.strategy.commercial_producer import CommercialStrategy
+from d3a.models.strategy.load_hours import LoadHoursStrategy
+
+
+def get_setup(config):
+    area = Area(
+        'Grid',
+        children=[
+            Area(
+                'House 1',
+                event_list=[DisconnectIntervalAreaEvent(6, 16)],
```

Comment:
Doesn't event_list=[DisconnectAreaEvent(12), ConnectAreaEvent(16)] makes event_list=[DisconnectIntervalAreaEvent(6, 16)] it redundant

Reply:
I vote for keeping it. And how about adding DisableInterval as well?

Reply:
Sorry, overlooked that. Thanks!

Reply:
Sorry, I have a general comment concerning the two types (interval and individual):
IMO the code in area.events is quite convoluted (had a hard time understanding it). Maybe we can get rid of the individual events? IMO an interval is more straight forward fo implement for the user.
What do you think?

Reply:
If it was just for the connect/disconnect and enable/disable I would definitely agree. But the story I am working on right now is about strategy events, in which case interval events do not make sense (the user will probably not be interested in reverting the parameters, just to see the result of the actual change), in which case we need the individual events.

Reply:
Ah ok, I see. Thanks!